pkg/controller/egress/validate.go

```diff
@@ -53,6 +54,31 @@ func (c *EgressController) ValidateEgress(review *admv1.AdmissionReview) *admv1.
 		if len(newEgress.Spec.ExternalIPPools) > 0 {
 			return false, "spec.externalIPPools is not supported yet"
 		}
+		// Validate Egress trafficShaping
+		if newEgress.Spec.Bandwidth != nil {
+			pattern := "^[1-9][0-9]*[KMG]$"
+			if matched, _ := regexp.MatchString(pattern, newEgress.Spec.Bandwidth.Rate); !matched {
+				return false, fmt.Sprintf("Rate %s in Egress %s is Invalid. Expected format is [integer + K(Kbps), M(Mbps) or G(Gbps)]. e.g. 300K, 100M, 2G", newEgress.Spec.Bandwidth.Rate, newEgress.Name)
+			}
+			if matched, _ := regexp.MatchString(pattern, newEgress.Spec.Bandwidth.Burst); !matched {
+				return false, fmt.Sprintf("Burst %s in Egress %s is Invalid. Expected format is [integer + K(Kbps), M(Mbps) or G(Gbps)]. e.g. 300K, 100M, 2G", newEgress.Spec.Bandwidth.Burst, newEgress.Name)
+			}
+		}
+		// Validate if an EgressIP used by a rate-limited Egress is shared.
+		if newEgress.Spec.EgressIP != "" {
+			objs, _ := c.egressInformer.Informer().GetIndexer().ByIndex(egressIPIndex, newEgress.Spec.EgressIP)
+			if newEgress.Spec.Bandwidth != nil && len(objs) > 0 {
+				if len(objs) > 1 || objs[0].(*crdv1beta1.Egress).Name != newEgress.Name {
+					return false, fmt.Sprintf("Egress with traffic-shaping can't share EgressIP %s with other Egresses", newEgress.Spec.EgressIP)
+				}
+			}
+			for _, obj := range objs {
+				egress := obj.(*crdv1beta1.Egress)
+				if egress.Spec.Bandwidth != nil && egress.Name != newEgress.Name {
+					return false, fmt.Sprintf("EgressIP %s used by other Egress %s with traffic-shaping can't be shared", newEgress.Spec.EgressIP, egress.Name)
+				}
+			}
+		}
```

Comment:
To be simpler, perhaps we can just document an Egress IP can be applied a bandwidth only. If multiple Egresses use the same IP but configure different bandwidths, the effective bandwidth will be selected randonly. Like https://github.com/antrea-io/antrea/blob/ae5e29b84e653837a603e982ef870e2dc8fe317b/docs/egress.md?plain=1#L120-L121

This makes it possible to share the same IP and the same bandwidth across multiple Egresses.

Reply:
Sounds good. Done.